File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.java
class Solution {
    public int[] createTargetArray(int[] nums, int[] index) {

        ArrayList<Integer> outputArr=new ArrayList<Integer>(nums.length);
        
        for(int i=0;i<nums.length;i++){
            outputArr.add(index[i],nums[i]);
        }
        for(int i=0;i<nums.length;i++){
            nums[i]=outputArr.get(i);
        }
        //modifing parameter arr because its pass by value        
        return nums;
    }
}

File: 1389-create-target-array-in-the-given-order/1389-create-target-array-in-the-given-order.py
class Solution:
    def createTargetArray(self, nums: List[int], index: List[int]) -> List[int]:
        outputList=[]#*len(nums)
        tempZip=zip(nums,index)
        for n,i in tempZip:
            outputList.insert(i,n)
            # print(n,i)
        return outputList